x/wasm/internal/keeper/cosmwasm/contract.go

```diff
@@ -0,0 +1,35 @@
+package cosmwasm
+
+import (
+	"encoding/json"
+
+	wasmTypes "github.com/CosmWasm/go-cosmwasm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type OnReceiveIBCResponse struct {
+	Messages []sdk.Msg `json:"messages"`
+
+	Acknowledgement json.RawMessage
+	// log message to return over abci interface
+	Log []wasmTypes.LogAttribute `json:"log"`
+}
+
+type OnAcknowledgeIBCResponse struct {
+	Messages []sdk.Msg `json:"messages"`
+
+	// log message to return over abci interface
+	Log []wasmTypes.LogAttribute `json:"log"`
+}
+type OnTimeoutIBCResponse struct {
+	Messages []sdk.Msg `json:"messages"`
+
+	// log message to return over abci interface
+	Log []wasmTypes.LogAttribute `json:"log"`
+}
+
+type AcceptChannelResponse struct {
```

Comment:
This would be one more contract callback? for the `OnTryInitChannel` or whatever it is called?

Reply:
I have used this for `OnChanOpenInit` `OnChanOpenTry` but channel handshake was only touched in this spike.
My idea was to have the contracts engaged here as well. For example if a contract is end-of-life (burned, escrow released,...) it would make sense to not accept any new channels.